### PR TITLE
feat: use graphql-in-server-components

### PR DESCRIPTION
**Pull request by JoviDeCroock**
Documented in #58 

@mxstbr I would love your review on stuff in `/packages/core`

Also wondering whether to stub in `headers` which would make the API consistent between server and client.

```
import { headers } from 'next/headers'
Object.defineProperty(ctx, 'headers' { get() { return headers() } })
```

### CODE REVIEW

File: examples/spacex/app/rsc/page.tsx [hunk 1]
```diff
@@ -37,7 +35,22 @@ export default async function Page({
...
+    document: LaunchesQuery,
+    schema: builder.toSchema(),
+    variableValues: {
+      limit: 10,
+      offset,
+    },
+    contextValue: {},
+  })
+
+  const newResult = (result.data.launches = {
```
Comment (JoviDeCroock) [inline examples/spacex/app/rsc/page.tsx:48]: We need to remove Null prototypes for Next.js

File: examples/spacex/app/rsc/page.tsx [hunk 2]
```diff
@@ -8,14 +9,11 @@ import styles from './page.module.css'
...
 
-const { getClient } = registerClient(() =>
-  createClient({
-    url:
-      process.env.NODE_ENV === 'production'
-        ? 'https://spacex-fuse.vercel.app/api/fuse'
-        : 'http://localhost:3000/api/fuse',
-  }),
-)
+const files = require.context('../../types', true, /\.ts$/)
```
Comment (JoviDeCroock) [inline examples/spacex/app/rsc/page.tsx:12]: Somehow we need to remove the `require.context` to not make this optimisation feel like a hurdle

**Review by TheMightyPenguin — APPROVED** [review]
This is dope 💯

**Review by mxstbr — APPROVED** [review]
Amazing!!!

**Pull request merged**